cosa/encoders/coreir.py

```diff
@@ -338,6 +338,8 @@ def extract_value(x, modname, inst_intr, inst_conf, inst_mod):
                     xval = 1 if xval else 0
                 else:
                     if type(xval) != int:
+                        if xval.is_x():
+                            return None
```

Comment:
The return None is not aligned with the rest of the code i.e., return xval. I would suggest to either set xval, or change all parts to locally return the value.